instruments/thorlabs/_packets.py

```diff
@@ -66,12 +64,17 @@ def __str__(self):
         return """
 ThorLabs APT packet:
     Message ID      0x{0._message_id:x}
-    Parameter 1     0x{0._param1:x}
-    Parameter 2     0x{0._param2:x}
+    Parameter 1     {1}
+    Parameter 2     {2}
     Destination     0x{0._dest:x}
     Source          0x{0._source:x}
-    Data            {1}
-""".format(self, "{:x}".format(self._data) if self._has_data else "None")
+    Data            {3}
+""".format(
+    self,
+    "0x{:x}".format(self._param1) if not self._has_data else "None",
+    "0x{:x}".format(self._param2) if not self._has_data else "None",
+    "{}".format(self._data) if self._has_data else "None"
+)
```

Comment:
Stiching together the string had some errors for parameters that were None and can't be treated as parameters. Now prints fine and has tests.